installer.sh

```diff
@@ -1,18 +1,88 @@
 #!/bin/bash
 
-# Dependency installation
-echo "Installing dependencies..."
-sudo apt-get update
-if ! sudo apt-get install -y python3-pip python3-gpiozero; then
-    echo "Failed to install system packages."
+# Rotary Phone Audio Guestbook Installer
+
+echo "Starting the installation process..."
+
+# Update and install system dependencies
+echo "Installing additional dependencies..."
+sudo apt-get install -y python3-pip python3-venv python3-gpiozero ffmpeg || {
+    echo "Failed to install required system packages."
     exit 1
-fi
+}
 
-# Use --user flag for pip installations
-if ! pip3 install --user pydub pyaudio PyYAML; then
-    echo "Failed to install Python packages."
+# Set up Python virtual environment for project dependencies
+echo "Setting up Python virtual environment..."
+python3 -m venv ~/rotary-phone-venv || {
+    echo "Failed to create Python virtual environment."
     exit 1
-fi
+}
+source ~/rotary-phone-venv/bin/activate
+
+# Install Python dependencies in the virtual environment
+pip install pydub pyaudio PyYAML sounddevice || {
```

Comment:
Unrelated to to this change, but for stability, I think you might want to lock in some specific versions in a `requirements.txt` file.

Reply:
done!